v3/src/components/graph/components/axis-attribute-menu.tsx

```diff
@@ -0,0 +1,61 @@
+import { Menu, MenuItem, MenuList, MenuButton, MenuDivider } from "@chakra-ui/react"
+import React, { CSSProperties } from "react"
+import { useDataSetContext } from "../../../hooks/use-data-set-context"
+import { AxisPlace, GraphPlace } from "../models/axis-model"
+import { useGraphLayoutContext} from "../models/graph-layout"
+import { measureText } from "../../../hooks/use-measure-text"
+
+import "./axis-attribute-menu"
+
+interface IProps {
+  attrId: string
+  place: GraphPlace,
+  onChangeAttribute: (place: string, attrid: string) => void
+  onTreatAs: (place: string, attrid: string, treatAs: string) => void
+}
+
+export const AxisAttributeMenu = ({ attrId, place, onChangeAttribute, onTreatAs }: IProps ) => {
+  const data = useDataSetContext()
+  const attribute = data?.attrFromID(attrId)
+  const treatAs = attribute?.type === "numeric" ? "categorical" : "numeric"
+  const layout = useGraphLayoutContext()
+  const { margin } = layout
+  const textLength = measureText(attribute?.name || "")
+
+  // TODO - replace with a more reliable calculation or way of placing button
+  const calcLeft = () => {
+    const bounds = layout.getAxisBounds(place as AxisPlace)
+    if (!bounds) return 0
+    return place === "left" ? -30 : (bounds.width * .5) - (textLength * .5) + margin.left - 5
+  }
```

Comment:
That is exactly what I would like to do, yes, but was not sure I could easily do it from where this component sits in the tree.  Wanted to check with you on the best way to achieve it, since this component has to be a peer of the Axis at the moment.